lib/matplotlib/tests/test_backend_svg.py

```diff
@@ -179,3 +179,22 @@ def psfont(*args, **kwargs):
     ax.text(0.5, 0.5, 'hello')
     with tempfile.TemporaryFile() as tmpfile, pytest.raises(ValueError):
         fig.savefig(tmpfile, format='svg')
+
+
+def test_NonGui_warning():
+    matplotlib.use('Agg')
```

Comment:
To force an alternate backend, you can mark the function with a decorator: `@pytest.mark.backend('Agg')`